torch/_inductor/cudagraph_trees.py

```diff
@@ -323,26 +323,23 @@ def is_cuda_tensor(x):
 
 
 @contextlib.contextmanager
-def _use_cuda_memory_pool_manager(device, mem_pool):
+def _use_cuda_memory_pool_manager(device, mem_pool, stream):
     """
     Context manager to use cuda graph pool for new allocations. If you use this manager
     all cudagraph tensors in use should be reflected in the allocator or they will be overwritten.
     existing_graph should already have been used in a capture, and the mem_pool must already exist,
     because this manager will not preserve a reference to the pool which keeps it alive.
     """
-    with torch.device(device):
-        torch.cuda.synchronize()
-        stream = torch.cuda.Stream()
-        stream.wait_stream(torch.cuda.current_stream())
-        stream_context = torch.cuda.stream(stream)
-        stream_context.__enter__()
+    torch.cuda.synchronize()
+    stream.wait_stream(torch.cuda.current_stream())
+
+    with torch.cuda.stream(stream), torch.device(device):
```

Comment:
what does stream(stream) do - create a new stream based on given stream?

Reply:
its a context manager for enabling the passed in stream